client/clientd/src/lib.rs

```diff
@@ -1,24 +1,49 @@
 use anyhow::Result;
+use async_trait::async_trait;
+use axum::extract::rejection::JsonRejection;
+use axum::extract::{FromRequest, RequestParts};
+use axum::response::{IntoResponse, Response};
+use axum::BoxError;
 use bitcoin::hashes::hex::ToHex;
-use fedimint_api::{Amount, OutPoint};
+use bitcoin::Transaction;
+use fedimint_api::{Amount, OutPoint, TransactionId};
 use fedimint_core::modules::mint::tiered::coins::Coins;
+use fedimint_core::modules::wallet::txoproof::TxOutProof;
 use mint_client::mint::{CoinFinalizationData, SpendableCoin};
+use mint_client::ClientError;
+use reqwest::StatusCode;
+use serde::de::DeserializeOwned;
 use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+use thiserror::Error;
 
-#[derive(Deserialize, Serialize)]
-pub enum RpcResult {
-    #[serde(rename = "success")]
-    Success(serde_json::Value),
-    #[serde(rename = "failure")]
-    Failure(serde_json::Value),
+#[derive(Error, Debug)]
+pub enum ClientdError {
+    #[error("Client error: {0}")]
+    ClientError(#[from] ClientError),
+}
+
+impl IntoResponse for ClientdError {
+    fn into_response(self) -> Response {
+        let payload = json!({ "error": self.to_string(), });
+        let code = StatusCode::BAD_REQUEST;
+        Result::<(), _>::Err((code, axum::Json(payload))).into_response()
+    }
```

Comment:
Did you look into best practices about how the error should be returned? Only in JSON? Also as status code?